hazelcast/src/test/java/com/hazelcast/map/impl/querycache/QueryCacheTest.java

```diff
@@ -230,6 +235,39 @@ public void getAll_with_non_existent_keys() {
         queryCache.getAll(keySet);
     }
 
+    @Test
+    public void testQueryCache_with_mapAttribute_inPredicate() {
+
+        String MAP_ATTRIBUTE_NAME = "booleanMapAttribute";
+
+        Config config = new Config();
+        config.getMapConfig(mapName)
+                .addQueryCacheConfig(
+                        new QueryCacheConfig(cacheName)
+                                .setIncludeValue(true)
+                                .setPredicateConfig(// use map attribute in a predicate
+                                        new PredicateConfig(Predicates.equal(MAP_ATTRIBUTE_NAME, true))
```

Comment:
Extractors on IMap side can work but when the same predicate is ran against query-cache, there will still be exceptions. This may be seen as inconsistent behavior. Are we ok with this?

Reply:
It will enough to push the query cache data to clients using map attributes. Then, at the client side, the query cache can be queried without map attributes or even via full table scan.